datadog_checks_downloader/tests/test_downloader.py

```diff
@@ -37,39 +64,289 @@
+
+        # No exception chaining done, make a check to see ExpiredMetadataError in the exception string.
+        assert "ExpiredMetadataError(\"Metadata 'timestamp' expired on" in str(exc)
+
+
+@pytest.mark.offline
+@pytest.mark.skipif(PY2, reason="tuf builds for Python 2 do not provide required information in exception")
+def test_local_unreachable_repository():
+    """Test unreachable repository raises an exception."""
+    argv = [
+        "datadog-some-distribution",
+        "--version",
+        "1.0.0",
+        "--repository",
+        "http://localhost:1",
+    ]
+
+    with pytest.raises(NoWorkingMirrorError) as exc:
+        _do_run_downloader(argv)
+
+    # No exception chaining done, check the exception content.
+    assert "ConnectionError(MaxRetryError(" in str(exc)
+
+
+@pytest.mark.offline
+@pytest.mark.parametrize(
+    "distribution_name,distribution_version",
+    [
+        ("datadog-active-directory", "1.10.0"),
+    ],
+)
+@pytest.mark.skipif(PY2, reason="tuf builds for Python 2 do not provide required information in exception")
+@mock.patch("time.time", mock.MagicMock(return_value=_LOCAL_TESTS_DATA_TIMESTAMP))
+def test_local_wheels_signer_signature_leaf_error(distribution_name, distribution_version):
+    """Test failure in verifying wheels-signer signature.
+
+    The wheel-signer-{a-z} metadata has to have wrong signature.
+    """
+    test_data = "{}-{}-signature-wheels-signer-a".format(distribution_name, distribution_version)
+    with local_http_server(test_data) as http_url:
+        argv = [
+            distribution_name,
+            "--version",
+            distribution_version,
+            "--repository",
+            http_url,
+        ]
+
+        with pytest.raises(NoWorkingMirrorError) as exc:
+            _do_run_downloader(argv)
+
+    assert "BadSignatureError('wheels-signer-" in str(exc)
+
+
 def delete_files(files):
     for f in files:
         os.remove(f)
 
 
-def cleanup():
-    REPO_DIR = 'datadog_checks/downloader/data/repo/'
+def _cleanup():
+    REPO_DIR = "datadog_checks/downloader/data/repo/"
 
-    METADATA_DIR = os.path.join(REPO_DIR, 'metadata')
-    TARGETS_DIR = os.path.join(REPO_DIR, 'targets')
-    IN_TOTO_METADATA_DIR = os.path.join(TARGETS_DIR, 'in-toto-metadata')
-    IN_TOTO_PUBKEYS_DIR = os.path.join(TARGETS_DIR, 'in-toto-pubkeys')
-    SIMPLE_DIR = os.path.join(TARGETS_DIR, 'simple')
+    METADATA_DIR = os.path.join(REPO_DIR, "metadata")
+    TARGETS_DIR = os.path.join(REPO_DIR, "targets")
+    IN_TOTO_METADATA_DIR = os.path.join(TARGETS_DIR, "in-toto-metadata")
+    IN_TOTO_PUBKEYS_DIR = os.path.join(TARGETS_DIR, "in-toto-pubkeys")
+    SIMPLE_DIR = os.path.join(TARGETS_DIR, "simple")
 
     # First, nuke all known targets. but not the directory for targets itself.
-    shutil.rmtree(IN_TOTO_METADATA_DIR)
-    shutil.rmtree(IN_TOTO_PUBKEYS_DIR)
-    shutil.rmtree(SIMPLE_DIR)
+    shutil.rmtree(IN_TOTO_METADATA_DIR, ignore_errors=True)
```

Comment:
Were there errors to begin with?

Reply:
Otherwise I see the following error when running the testsuite:

```
============================================================================== FAILURES ==============================================================================
__________________________________________________________________________ test_downloader ___________________________________________________________________________
/Users/fridolin.pokorny/git/dd/integrations-core/datadog_checks_downloader/tests/test_downloader.py:459: in test_downloader
    _cleanup()
/Users/fridolin.pokorny/git/dd/integrations-core/datadog_checks_downloader/tests/test_downloader.py:332: in _cleanup
    shutil.rmtree(IN_TOTO_METADATA_DIR)
/Users/fridolin.pokorny/.pyenv/versions/3.8.13/lib/python3.8/shutil.py:709: in rmtree
    onerror(os.lstat, path, sys.exc_info())
/Users/fridolin.pokorny/.pyenv/versions/3.8.13/lib/python3.8/shutil.py:707: in rmtree
    orig_st = os.lstat(path)
E   FileNotFoundError: [Errno 2] No such file or directory: 'datadog_checks/downloader/data/repo/targets/in-toto-metadata'
```